tests/python/unittest/test_lang_tensor_overload_op.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import tvm
+
+
+def test_bop_without_topi():
+    n = tvm.var('n')
+    m = tvm.var('m')
+    A = tvm.placeholder((n,), name='A')
+    B = tvm.placeholder((m,), name='B')
+    """
+    If topi, which provides overloaded operator support, is not imported,
+    ValueError should be thrown.
+    """
+    try:
+        C = A + B
+        assert False
+    except (ValueError, tvm.TVMError):
+        pass
+
+
+def test_operator_type():
+    import topi
+
+    k = 1
+    n = tvm.var('n')
+    A = tvm.placeholder((), name='A')
+    B = tvm.placeholder((10, 5), name='B')
+    B1 = B[0]
+    B2 = B[0,0]
+
+    assert isinstance(k + n, tvm.expr.Expr)
+    assert isinstance(n + n, tvm.expr.Expr)
+    assert isinstance(k + A, tvm.expr.Expr)
+    assert isinstance(A + k, tvm.expr.Expr)
+    assert isinstance(n + A, tvm.expr.Expr)
+    assert isinstance(A + n, tvm.expr.Expr)
+    assert isinstance(A + A, tvm.expr.Expr)
+
+    assert isinstance(k + B, tvm.tensor.Tensor)
+    assert isinstance(B + k, tvm.tensor.Tensor)
+    assert isinstance(n + B, tvm.tensor.Tensor)
+    assert isinstance(B + n, tvm.tensor.Tensor)
+    assert isinstance(A + B, tvm.tensor.Tensor)
+    assert isinstance(B + A, tvm.tensor.Tensor)
+    assert isinstance(B + B, tvm.tensor.Tensor)
+
+    assert isinstance(k + B2, tvm.expr.Expr)
+    assert isinstance(B2 + k, tvm.expr.Expr)
+    assert isinstance(n + B2, tvm.expr.Expr)
+    assert isinstance(B2 + n, tvm.expr.Expr)
+    assert isinstance(B2 + B2, tvm.expr.Expr)
+    assert isinstance(B2 + A, tvm.expr.Expr)
+    assert isinstance(A + B2, tvm.expr.Expr)
+    assert isinstance(B2 + B, tvm.tensor.Tensor)
+    assert isinstance(B + B2, tvm.tensor.Tensor)
+
+
+def test_tensor_scalar_bop():
+    import topi
+
+    def tensor_scalar_bop(op):
+        n = tvm.var('n')
+        k = tvm.var('k')
+        A = tvm.placeholder((n,), name='A')
+        if op == "add":
+            B = A + k
+        elif op == "sub":
+            B = A - k
+        elif op == "mul":
+            B = A * k
+        elif op == "div":
+            B = A / k
+        s = tvm.create_schedule(B.op)
+        f = tvm.build(s, [n, k, A, B], "llvm")
+
+        ctx = tvm.cpu(0)
+        n = 10
+        k = 5
+        a = tvm.nd.array(np.random.uniform(size=n).astype(A.dtype), ctx)
+        b = tvm.nd.array(np.zeros(n, dtype=B.dtype), ctx)
+        f(n, k, a, b)
+
+        if op == "add":
+            output = a.asnumpy() + k
+        elif op == "sub":
+            output = a.asnumpy() - k
+        elif op == "mul":
+            output = a.asnumpy() * k
+        elif op == "div":
+            output = a.asnumpy() / k
+        np.testing.assert_allclose(b.asnumpy(), output)
+
```

Comment:
add testcase for other device types

Reply:
add a testcase for tensor constant array ops float

Reply:
I don't understand "by tensor constant array ops float". Do you mean operator of Tensor and array<float>? Something like:
`A = tvm.placeholder((5, 10), name='A')`
`B = A + [1.0, 2.0, 3.0]        # operator of Tensor and array<float>`

I don't know how to do this. Because I would try creating a tmp Tensor with the shape of the array (i.e., shape=(3,) in this case), and returning broadcast_add(A, tmp). In this case, the constants 1.0, 2.0, and 3.0 are not preserved, unlike Tensor-constant_scalar ops where the constant_scalar is preserved.

Would you shed some light on how to do this, please?